node/src/test/java/io/airlift/node/TestNodeConfig.java

```diff
@@ -45,12 +47,14 @@ public void testDefaults()
                 .setLocation(null)
                 .setBinarySpec(null)
                 .setConfigSpec(null)
-                .setInternalAddressSource(IP));
+                .setInternalAddressSource(IP)
+                .setAnnotationFile(null));
     }
 
     @Test
     public void testExplicitPropertyMappings()
     {
+        File annotationFile = new File(requireNonNull(getClass().getClassLoader().getResource("annotations.properties")).getFile());
```

Comment:
same here

> requireNonNull -> use guava `Resources`
`URL.getFile()` -> this will pass for resource URLs like `jar:...`, but you want it to fail

> use `new File(url.toURI())`